### PR TITLE
Endpoints to update/delete a jurisdiction

### PR DESCRIPTION
**Pull request by drtyh2o**
Ent

### CODE REVIEW

File: packages/mds-jurisdiction-service/index.ts [hunk 1]
```diff
@@ -111,6 +112,71 @@ const createJurisdiction = async (
...
+            raw: [updated]
+          }: UpdateReturning<JurisdictionEntity> = await connection
+            .getRepository(JurisdictionEntity)
+            .createQueryBuilder()
+            .update()
+            .set({
+              ...update,
+              agency_key: updates.agency_key ?? entity?.agency_key,
+              versions:
+                (updates.agency_name ?? current.agency_name) !== current.agency_name ||
```
Comment (macsj200) [inline packages/mds-jurisdiction-service/index.ts]: There is _a lot_ of inline logic + syntax here, `??`, `...`, etc.

File: packages/mds-jurisdiction-service/index.ts [hunk 2]
```diff
@@ -111,6 +112,118 @@ const createJurisdiction = async (
...
+        .where({ jurisdiction_id })
+        .getOne()
+      if (entity) {
+        const current = AsJurisdiction()(entity)
+        if (current) {
+          const { id, ...update } = entity
+          await connection
+            .getRepository(JurisdictionEntity)
+            .createQueryBuilder()
+            .update()
```
Comment (janedotx) [inline packages/mds-jurisdiction-service/index.ts]: Why is this `.update` and not `.delete` if this is meant to delete a Jurisdiction?

Comment (drtyh2o) [reply packages/mds-jurisdiction-service/index.ts]: Using `.update` to _soft_ delete a jurisdiction.

File: packages/mds-jurisdiction-service/tests/index.spec.ts [hunk 3]
```diff
@@ -106,25 +106,71 @@ describe('Write/Read Jurisdictions', () => {
...
-  //     effective: YESTERDAY
-  //   })
-  //   test.value(jurisdiction).isNot(null)
-  //   test.value(jurisdiction?.jurisdiction_id).is(JURISDICTION_ID)
-  //   test.value(jurisdiction?.timestamp).is(YESTERDAY)
-  //   test.value(error).is(null)
-  // })
+  it('Read Specific Jurisdiction (prior version)', async () => {
+    const [error, jurisdiction] = await JurisdictionService.getOneJurisdiction(JURISDICTION_ID, {
+      effective: YESTERDAY
```
Comment (janedotx) [inline packages/mds-jurisdiction-service/tests/index.spec.ts]: I don't understand why you'd ever want to pass in the timestamp. Is a `jurisdiction_id` not uniquely identifying?

Comment (drtyh2o) [reply packages/mds-jurisdiction-service/tests/index.spec.ts]: It is uniquely identifying and has to be ... however, jurisdictions can have versions that change over time.

File: packages/mds-jurisdiction-service/index.ts (hunk 2 above)
Comment (janedotx) [reply packages/mds-jurisdiction-service/index.ts]: Never mind, I read the spec.

File: packages/mds-jurisdiction-service/tests/index.spec.ts (hunk 3 above)
Comment (janedotx) [reply packages/mds-jurisdiction-service/tests/index.spec.ts]: Never mind, read the spec.

File: packages/mds-jurisdiction-service/index.ts (hunk 2 above)
Comment (drtyh2o) [reply packages/mds-jurisdiction-service/index.ts]: Adding a comment to clarify the behavior.

File: packages/mds-jurisdiction-service/index.ts [hunk 4]
```diff
@@ -111,25 +99,107 @@ const createJurisdiction = async (
...
+  update: UpdateJurisdictionType
+): Promise<JurisdictionServiceResult<Jurisdiction, ValidationError | NotFoundError>> => {
+  if (update.jurisdiction_id && update.jurisdiction_id !== jurisdiction_id) {
+    return Failure(new ValidationError('Invalid jurisdiction_id for update'))
+  }
+  try {
+    try {
+      const entity = await orm.readJurisdiction(jurisdiction_id)
+      if (entity) {
+        const current = AsJurisdiction()(entity)
```
Comment (avatarneil) [inline packages/mds-jurisdiction-service/index.ts]: I love currying

**Pull request merged**